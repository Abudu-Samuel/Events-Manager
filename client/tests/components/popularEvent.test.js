import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import PopularEvents from '../../components/Event/PopularCenter';
import eventMockedData from '../__mocks__/eventMockedData';


const props = {
  events: {
    event: [eventMockedData.eventData]
  }
};

describe('<PopularEvents />', () => {
  it('should render Footer correctly', () => {
    const wrapper = shallow(<PopularEvents {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
