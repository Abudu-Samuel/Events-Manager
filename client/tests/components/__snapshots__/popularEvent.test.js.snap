// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopularEvents /> should render Footer correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopularEvents
    events={
      Object {
        "event": Array [
          Object {
            "date": "2018-12-06",
            "description": "awesome",
            "image": "image.png",
            "title": "evento",
            "type": "wedding",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        id="events"
      >
        <h2
          className="mb-3 slot text-center font-weight-bold"
        >
          Latest Events
        </h2>
        <div
          className="row mb-4"
        >
          <div
            className="col-md-4 mb-4"
          >
            <div
              className="card hoverable text-center"
            >
              <img
                className="img-fluid hoverable max"
                src="image.png"
              />
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                   
                  evento
                </h4>
                <p
                  className="card-text"
                >
                  awesome
                </p>
                <p
                  className="card-text"
                >
                  Date: 
                  2018-12-06
                </p>
                <button
                  className="btn btn-mycolor btn-sm"
                  id="viewEvent"
                >
                  <Link
                    replace={false}
                    to="/events/undefined"
                  >
                    Details
                  </Link>
                </button>
              </div>
            </div>
          </div>
        </div>
      </section>,
      "className": "container space",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="mb-3 slot text-center font-weight-bold"
          >
            Latest Events
          </h2>,
          <div
            className="row mb-4"
          >
            <div
              className="col-md-4 mb-4"
            >
              <div
                className="card hoverable text-center"
              >
                <img
                  className="img-fluid hoverable max"
                  src="image.png"
                />
                <div
                  className="card-body"
                >
                  <h4
                    className="card-title"
                  >
                     
                    evento
                  </h4>
                  <p
                    className="card-text"
                  >
                    awesome
                  </p>
                  <p
                    className="card-text"
                  >
                    Date: 
                    2018-12-06
                  </p>
                  <button
                    className="btn btn-mycolor btn-sm"
                    id="viewEvent"
                  >
                    <Link
                      replace={false}
                      to="/events/undefined"
                    >
                      Details
                    </Link>
                  </button>
                </div>
              </div>
            </div>
          </div>,
        ],
        "id": "events",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Latest Events",
            "className": "mb-3 slot text-center font-weight-bold",
          },
          "ref": null,
          "rendered": "Latest Events",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-md-4 mb-4"
              >
                <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="image.png"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                       
                      evento
                    </h4>
                    <p
                      className="card-text"
                    >
                      awesome
                    </p>
                    <p
                      className="card-text"
                    >
                      Date: 
                      2018-12-06
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewEvent"
                    >
                      <Link
                        replace={false}
                        to="/events/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>
              </div>,
            ],
            "className": "row mb-4",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="image.png"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                       
                      evento
                    </h4>
                    <p
                      className="card-text"
                    >
                      awesome
                    </p>
                    <p
                      className="card-text"
                    >
                      Date: 
                      2018-12-06
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewEvent"
                    >
                      <Link
                        replace={false}
                        to="/events/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>,
                "className": "col-md-4 mb-4",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      className="img-fluid hoverable max"
                      src="image.png"
                    />,
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                         
                        evento
                      </h4>
                      <p
                        className="card-text"
                      >
                        awesome
                      </p>
                      <p
                        className="card-text"
                      >
                        Date: 
                        2018-12-06
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewEvent"
                      >
                        <Link
                          replace={false}
                          to="/events/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>,
                  ],
                  "className": "card hoverable text-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "img-fluid hoverable max",
                      "src": "image.png",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h4
                          className="card-title"
                        >
                           
                          evento
                        </h4>,
                        <p
                          className="card-text"
                        >
                          awesome
                        </p>,
                        <p
                          className="card-text"
                        >
                          Date: 
                          2018-12-06
                        </p>,
                        <button
                          className="btn btn-mycolor btn-sm"
                          id="viewEvent"
                        >
                          <Link
                            replace={false}
                            to="/events/undefined"
                          >
                            Details
                          </Link>
                        </button>,
                      ],
                      "className": "card-body",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            " ",
                            "evento",
                          ],
                          "className": "card-title",
                        },
                        "ref": null,
                        "rendered": Array [
                          " ",
                          "evento",
                        ],
                        "type": "h4",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "awesome",
                          "className": "card-text",
                        },
                        "ref": null,
                        "rendered": "awesome",
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "Date: ",
                            "2018-12-06",
                          ],
                          "className": "card-text",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Date: ",
                          "2018-12-06",
                        ],
                        "type": "p",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Link
                            replace={false}
                            to="/events/undefined"
                          >
                            Details
                          </Link>,
                          "className": "btn btn-mycolor btn-sm",
                          "data-centerid": undefined,
                          "id": "viewEvent",
                          "onClick": undefined,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "Details",
                            "replace": false,
                            "to": "/events/undefined",
                          },
                          "ref": null,
                          "rendered": "Details",
                          "type": [Function],
                        },
                        "type": "button",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          id="events"
        >
          <h2
            className="mb-3 slot text-center font-weight-bold"
          >
            Latest Events
          </h2>
          <div
            className="row mb-4"
          >
            <div
              className="col-md-4 mb-4"
            >
              <div
                className="card hoverable text-center"
              >
                <img
                  className="img-fluid hoverable max"
                  src="image.png"
                />
                <div
                  className="card-body"
                >
                  <h4
                    className="card-title"
                  >
                     
                    evento
                  </h4>
                  <p
                    className="card-text"
                  >
                    awesome
                  </p>
                  <p
                    className="card-text"
                  >
                    Date: 
                    2018-12-06
                  </p>
                  <button
                    className="btn btn-mycolor btn-sm"
                    id="viewEvent"
                  >
                    <Link
                      replace={false}
                      to="/events/undefined"
                    >
                      Details
                    </Link>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </section>,
        "className": "container space",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="mb-3 slot text-center font-weight-bold"
            >
              Latest Events
            </h2>,
            <div
              className="row mb-4"
            >
              <div
                className="col-md-4 mb-4"
              >
                <div
                  className="card hoverable text-center"
                >
                  <img
                    className="img-fluid hoverable max"
                    src="image.png"
                  />
                  <div
                    className="card-body"
                  >
                    <h4
                      className="card-title"
                    >
                       
                      evento
                    </h4>
                    <p
                      className="card-text"
                    >
                      awesome
                    </p>
                    <p
                      className="card-text"
                    >
                      Date: 
                      2018-12-06
                    </p>
                    <button
                      className="btn btn-mycolor btn-sm"
                      id="viewEvent"
                    >
                      <Link
                        replace={false}
                        to="/events/undefined"
                      >
                        Details
                      </Link>
                    </button>
                  </div>
                </div>
              </div>
            </div>,
          ],
          "id": "events",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Latest Events",
              "className": "mb-3 slot text-center font-weight-bold",
            },
            "ref": null,
            "rendered": "Latest Events",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col-md-4 mb-4"
                >
                  <div
                    className="card hoverable text-center"
                  >
                    <img
                      className="img-fluid hoverable max"
                      src="image.png"
                    />
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                         
                        evento
                      </h4>
                      <p
                        className="card-text"
                      >
                        awesome
                      </p>
                      <p
                        className="card-text"
                      >
                        Date: 
                        2018-12-06
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewEvent"
                      >
                        <Link
                          replace={false}
                          to="/events/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>
                  </div>
                </div>,
              ],
              "className": "row mb-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="card hoverable text-center"
                  >
                    <img
                      className="img-fluid hoverable max"
                      src="image.png"
                    />
                    <div
                      className="card-body"
                    >
                      <h4
                        className="card-title"
                      >
                         
                        evento
                      </h4>
                      <p
                        className="card-text"
                      >
                        awesome
                      </p>
                      <p
                        className="card-text"
                      >
                        Date: 
                        2018-12-06
                      </p>
                      <button
                        className="btn btn-mycolor btn-sm"
                        id="viewEvent"
                      >
                        <Link
                          replace={false}
                          to="/events/undefined"
                        >
                          Details
                        </Link>
                      </button>
                    </div>
                  </div>,
                  "className": "col-md-4 mb-4",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <img
                        className="img-fluid hoverable max"
                        src="image.png"
                      />,
                      <div
                        className="card-body"
                      >
                        <h4
                          className="card-title"
                        >
                           
                          evento
                        </h4>
                        <p
                          className="card-text"
                        >
                          awesome
                        </p>
                        <p
                          className="card-text"
                        >
                          Date: 
                          2018-12-06
                        </p>
                        <button
                          className="btn btn-mycolor btn-sm"
                          id="viewEvent"
                        >
                          <Link
                            replace={false}
                            to="/events/undefined"
                          >
                            Details
                          </Link>
                        </button>
                      </div>,
                    ],
                    "className": "card hoverable text-center",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "img-fluid hoverable max",
                        "src": "image.png",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "img",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <h4
                            className="card-title"
                          >
                             
                            evento
                          </h4>,
                          <p
                            className="card-text"
                          >
                            awesome
                          </p>,
                          <p
                            className="card-text"
                          >
                            Date: 
                            2018-12-06
                          </p>,
                          <button
                            className="btn btn-mycolor btn-sm"
                            id="viewEvent"
                          >
                            <Link
                              replace={false}
                              to="/events/undefined"
                            >
                              Details
                            </Link>
                          </button>,
                        ],
                        "className": "card-body",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              " ",
                              "evento",
                            ],
                            "className": "card-title",
                          },
                          "ref": null,
                          "rendered": Array [
                            " ",
                            "evento",
                          ],
                          "type": "h4",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "awesome",
                            "className": "card-text",
                          },
                          "ref": null,
                          "rendered": "awesome",
                          "type": "p",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              "Date: ",
                              "2018-12-06",
                            ],
                            "className": "card-text",
                          },
                          "ref": null,
                          "rendered": Array [
                            "Date: ",
                            "2018-12-06",
                          ],
                          "type": "p",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <Link
                              replace={false}
                              to="/events/undefined"
                            >
                              Details
                            </Link>,
                            "className": "btn btn-mycolor btn-sm",
                            "data-centerid": undefined,
                            "id": "viewEvent",
                            "onClick": undefined,
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {
                              "children": "Details",
                              "replace": false,
                              "to": "/events/undefined",
                            },
                            "ref": null,
                            "rendered": "Details",
                            "type": [Function],
                          },
                          "type": "button",
                        },
                      ],
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
